shell.nix

```
@@ -1,11 +1,27 @@
-# Copyright (C) 2005 - 2021 Settlers Freaks <sf-team at siedler25.org>
+# Copyright (C) 2005 - 2022 Settlers Freaks <sf-team at siedler25.org>
 #
 # SPDX-License-Identifier: GPL-2.0-or-later
 
-{ pkgs ? import <nixpkgs> {} }:
+{
+  pkgs ? import <nixpkgs> { overlays = [
+    # This is a workaround for https://github.com/NixOS/nixpkgs/issues/146759
+    # && https://github.com/NixOS/nixpkgs/pull/149601 .
+    # NixOS does currently ship an incomple SDL2 installation
+    # when withStatic is set to false, see the issue for more info.
+    # The overlay can be safely removed as soon as the PR hits the targeted
+    # OS channels, see https://nixpk.gs/pr-tracker.html?pr=149601 as state overview.
+    # nixos-unstable should be a reasonable target, when we could consider
+    # removing this workaround
```

The issue link is enough for the reference/explanation. Let's keep this short, e.g.:

```suggestion
    # This is a workaround for https://github.com/NixOS/nixpkgs/issues/146759 :
    # NixOS does currently ship an incomplete SDL2 installation when withStatic=false.
    # nixos-unstable should be a reasonable target, to remove this workaround.
    # See https://nixpk.gs/pr-tracker.html?pr=149601 for a state overview.
```

---

I have shortened the text and split the commits, both makes absolute sense.